eventmesh-connectors/eventmesh-connector-chatgpt/src/main/java/org/apache/eventmesh/connector/chatgpt/source/managers/OpenaiManager.java

```diff
@@ -0,0 +1,124 @@
+import org.apache.eventmesh.common.utils.JsonUtils;
+import org.apache.eventmesh.connector.chatgpt.source.config.ChatGPTSourceConfig;
+import org.apache.eventmesh.connector.chatgpt.source.config.OpenaiConfig;
+import org.apache.eventmesh.connector.chatgpt.source.config.OpenaiProxyConfig;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.time.Duration;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.theokanning.openai.client.OpenAiApi;
+import com.theokanning.openai.completion.chat.ChatCompletionRequest;
+import com.theokanning.openai.completion.chat.ChatCompletionRequest.ChatCompletionRequestBuilder;
+import com.theokanning.openai.completion.chat.ChatMessage;
+import com.theokanning.openai.service.OpenAiService;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+
+
+@Slf4j
+public class OpenaiManager {
+
+    @Getter
+    private OpenAiService openAiService;
+
+    private String chatCompletionRequestTemplateStr;
+
+    public OpenaiManager(ChatGPTSourceConfig sourceConfig) {
+        initOpenAi(sourceConfig);
+    }
+
+    public String getResult(ChatCompletionRequest req) {
+        StringBuilder gptData = new StringBuilder();
+        try {
+            openAiService.createChatCompletion(req).getChoices()
+                .forEach(chatCompletionChoice -> gptData.append(chatCompletionChoice.getMessage().getContent()));
+        } catch (Exception e) {
+            log.error("Failed to generate GPT connection record: {}", e.getMessage());
+        }
+        return gptData.toString();
+    }
+
+    public ChatCompletionRequest newChatCompletionRequest(List<ChatMessage> chatMessages) {
+        ChatCompletionRequest request = JsonUtils.parseObject(chatCompletionRequestTemplateStr, ChatCompletionRequest.class);
+        request.setMessages(chatMessages);
+        return request;
+    }
+
+    private void initOpenAi(ChatGPTSourceConfig sourceConfig) {
+        OpenaiConfig openaiConfig = sourceConfig.getOpenaiConfig();
+        AssertUtils.isTrue(openaiConfig.getTimeout() > 0, "openaiTimeout must be >= 0");
+        boolean proxyEnable = sourceConfig.connectorConfig.isProxyEnable();
+        if (proxyEnable) {
+            OpenaiProxyConfig chatgptProxyConfig = sourceConfig.openaiProxyConfig;
+            if (chatgptProxyConfig.getHost() == null) {
+                throw new IllegalStateException("chatgpt proxy config 'host' cannot be null");
+            }
+            ObjectMapper mapper = defaultObjectMapper();
+            Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(chatgptProxyConfig.getHost(), chatgptProxyConfig.getPort()));
+            OkHttpClient client =
+                OpenAiService.defaultClient(openaiConfig.getToken(), Duration.ofSeconds(openaiConfig.getTimeout())).newBuilder().proxy(proxy).build();
+            Retrofit retrofit = defaultRetrofit(client, mapper);
```

Comment:
Can `OpenAiService.defaultClient()` be consistent with the following to use static import?